### ISSUE TITLE
DeckGL + VueJS update triggers not working

### ISSUE THREAD

**Issue by FilipHusnjak** [visgl/deck.gl#4826]
Hello,

I'm trying to implement feature to highlight objects when hovered in DeckGL using VueJS.
Function ```hover``` triggers properly when object is hovered but although that function changes variable ```selected```, ```getColor``` method is not invoked again to recalculate color.
Im using ```SimpleMeshLayer``` :
```
new SimpleMeshLayer({
          id: "building",
          opacity: 1.0,
          coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
          coordinateOrigin: [
            this.buildingViewState.longitude,
            this.buildingViewState.latitude
          ],
          data: [
            {
              position: [0, 0],
              color: [255, 255, 255],
              orientation: [0, 0, 90]
            }
          ],
          pickable: true,
          onHover: info => this.hover(info),
          mesh: this.MESH_URL_BUILDING,
          getPosition: d => d.position,
          getColor: d => {
            console.log(d.color);
            console.log(selected);
            return selected == 0 ? [255, 0, 0] : [255, 255, 255];
          },
          getOrientation: d => d.orientation,
          updateTriggers: {
            getColor: [selected]
          },
          getScale: [0.13, 0.13, 0.13]
        })
```
Function ```hover```:
```
hover(info) {
      if (info.picked) {
        selected = info.index;
      } else {
        selected = -1;
      }
      console.log(selected);
    }
```
```selected``` is global variable.

**Comment by Pessimistress** [comment]
You have to construct a new layer instance to update the trigger value. Any global variable is evaluated when it is used. You are effectively doing 

```js
let a = 1;
const b = a + 1;
a = 2;
```

And expecting b to update with a.

FYI you can achieve the same effect by using the `autoHighlight` and `highlightColor` props.

**Comment by FilipHusnjak** [comment]
Thank you,

I misunderstood the documentation for ```updateTriggers``` method.
I started using ```autoHighlight``` and ```highlightColor``` props and ran into a possible bug.
When I set ```highlightColor``` property to the ```SimpleMeshLayer``` object inside ```mounted()``` method in VueJS it is ignored and default highlight color is used.
Meanwhile when I update the ```Deck``` object layers with the new ```SimpleMeshLayer``` with the same properties inside onClick listener then correct highlight color is used. It also works with ```onHover``` listener.
